resources/assets/coffee/react/_components/tracklist.coffee

```diff
@@ -0,0 +1,196 @@
+        div className: "tracklisting__float-right tracklisting__voting-star#{if @props.track.selected then ' tracklisting__voting-star--selected' else ''}", href: '#', onClick: @handleClick,
+          i className: "fa fa-fw fa-refresh"
+      else
+        a className: "tracklisting__float-right tracklisting__voting-star#{if @props.track.selected then ' tracklisting__voting-star--selected' else ''}", href: '#', onClick: @handleClick,
+          i className: "fa fa-fw fa-star"
+
+class @TrackVoteSummary extends React.Component
+  render: ->
+    voteSummary = []
+    voteSummary.push _.times Math.max(0, @props.maxVotes - @props.voteCount), ->
+      div className: "tracklisting__float-right tracklisting__voting-star tracklisting__voting-star--smaller",
+        i className: "fa fa-fw fa-star"
+    voteSummary.push _.times @props.voteCount, ->
+      div className: "tracklisting__float-right tracklisting__voting-star tracklisting__voting-star--smaller tracklisting__voting-star--selected",
+        i className: "fa fa-fw fa-star"
+
+    div {},
+      voteSummary
+
+class @Track extends React.Component
+  playPreview: (e) =>
+    e.preventDefault()
+    $.publish 'tracklist:preview', id: @props.track.id
+
+  playDone: (e) =>
+    $.publish 'tracklist:previewDone', id: @props.track.id
+
+  render: ->
+    tr className: "tracklisting__row#{if @props.track.selected then ' tracklisting__row--selected' else ''}",
+      td className: 'tracklisting__cover', style: { backgroundImage: "url('#{@props.track.cover_url}')" },
+        a className: 'tracklisting__preview', href: '#', onClick: @playPreview,
+          i className: "fa fa-fw #{if @props.playing then 'fa-pause' else 'fa-play'}"
+          audio id: "track-#{@props.track.id}-audio", src: @props.track.preview, preload: 'none', onEnded: @playDone
+      td className:'tracklisting__title',
+        "#{@props.track.title} "
+        span className: 'tracklisting__version', @props.track.version
+      unless @props.options.hideBPM
+        td className:'tracklisting__bpm', "#{@props.track.bpm}bpm"
+      unless @props.options.hideGenre
+        td className:'tracklisting__genre', @props.track.genre
+      unless @props.options.hideDL
+        td className:'tracklisting__dl',
+          a className: 'tracklisting__link', href: @props.track.osz,
+            i className: 'fa fa-fw fa-cloud-download'
+      if @props.options.showVote
+        td className:'tracklisting__vote',
+          el TrackVoter, key: @props.track.id, track: @props.track, waitingForResponse: @props.waitingForResponse, voteCount: @props.voteCount, maxVotes: @props.options.maxVotes
+
+class @Tracklist extends React.Component
+
+  constructor: (props) ->
+    super props
+    @state =
+      waitingForResponse: false
+      currently_playing: null
+      tracks: @props.tracks
+      voteCount: _.filter(@props.tracks, _.iteratee({selected: true})).length
+      options:
+        hideBPM: @props.options.hideBPM ?= false
```

Comment:
this will change the props as well. Perhaps you mean just `@props.options.hideBPM ? false`?